### PR TITLE
feat(status-icons): update icons

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-799

- Update `StatusChip` and `StatusLabel` components with new icons
- Clean up useless icons
- Update tailwind
---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery  preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery  preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> The backlog never ends, be at peace with it

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [\[PR\] staging - feat(status-icons): update icons - 2023-09-15T13:16:50Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/8edf3258-b3bc-4a0d-a1ac-e46402222b80/services/general)

 Another comment will be posted when deployments are terminated

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/shared/ui/src/lib/components/icon/icons-status/spinner.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { type ReactNode, forwardRef } from 'react'
+import { twMerge } from 'tailwind-merge'
+import { type IconProps } from '../icon'
+
+type IconSpinnerProps = IconProps & {
+  children?: ReactNode
+}
```
Comment (ctjhoa) [inline libs/shared/ui/src/lib/components/icon/icons-status/spinner.tsx:7]: This should not be needed

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/shared/ui/src/lib/components/icon/icons-status/deployed.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { type ElementRef, forwardRef } from 'react'
+import { twMerge } from 'tailwind-merge'
+import { type IconStatusProps } from '../icon'
+
+export const DeployedIcon = forwardRef<ElementRef<'div'>, IconStatusProps>(
+  ({ className = '', width = 16, height = 16 }, forwardedRef) => {
```
[the remark this answers is not in the document]
Comment (ctjhoa) [reply libs/shared/ui/src/lib/components/icon/icons-status/deployed.tsx:6]: Still without name

File: libs/shared/ui/src/lib/components/icon/icons-status/spinner.tsx (hunk 1 above)
Comment (RemiBonnet) [reply libs/shared/ui/src/lib/components/icon/icons-status/spinner.tsx:10]: It should be necessary if we want to pass a children to Spinner component

File: libs/shared/ui/src/lib/components/icon/icons-status/canceled.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import { twMerge } from 'tailwind-merge'
+import { type IconStatusProps } from '../icon'
+
+export const CanceledIcon = forwardRef<ElementRef<'div'>, IconStatusProps>(function CanceledIcon(
+  { className = '', width = 16, height = 16 },
+  forwardedRef
+) {
+  return (
+    <div className={twMerge('text-neutral-300', className)} ref={forwardedRef}>
+      <svg xmlns="http://www.w3.org/2000/svg" width={width} height={height} fill="none" viewBox="0 0 24 24">
```
[the remark this answers is not in the document]
Comment (RemiBonnet) [reply libs/shared/ui/src/lib/components/icon/icons-status/canceled.tsx:11]: To be ISO with spinner component, but I can remove it for this kind of component

[the remark this answers is not in the document]
Comment (RemiBonnet) [reply libs/shared/ui/src/lib/components/icon/icons-status/canceled.tsx]: Fixed

File: libs/shared/ui/src/lib/components/icon/icons-status/deployed.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import { type ElementRef, forwardRef } from 'react'
+import { twMerge } from 'tailwind-merge'
+import { type IconStatusProps } from '../icon'
+
+export const DeployedIcon = forwardRef<ElementRef<'div'>, IconStatusProps>(
```
[the remark this answers is not in the document]
Comment (RemiBonnet) [reply libs/shared/ui/src/lib/components/icon/icons-status/deployed.tsx]: Not sure about it? why `ComponentPropsWithoutRef`?

File: libs/shared/ui/src/lib/components/icon/icons-status/build-error.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import { forwardRef } from 'react'
+import { twMerge } from 'tailwind-merge'
+import { type IconSVGProps } from '../icon'
+
+export const BuildErrorIcon = forwardRef<SVGSVGElement, IconSVGProps>(function BuildErrorIcon(
```
Comment (ctjhoa) [inline libs/shared/ui/src/lib/components/icon/icons-status/build-error.tsx:5]:
```suggestion
export const BuildErrorIcon = forwardRef<SVGElement, IconSVGProps>(function BuildErrorIcon(
```
https://developer.mozilla.org/en-US/docs/Web/API/SVGSVGElement
https://developer.mozilla.org/en-US/docs/Web/API/SVGElement

File: libs/shared/ui/src/lib/components/icon/icons-status/canceling.tsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import { type ElementRef, forwardRef } from 'react'
+import { twMerge } from 'tailwind-merge'
+import { type IconSVGProps } from '../icon'
+import SpinnerIcon from './spinner'
+
+export const CancelingIcon = forwardRef<ElementRef<'div'>, IconSVGProps>(function CancelingIcon(
```
Comment (ctjhoa) [inline libs/shared/ui/src/lib/components/icon/icons-status/canceling.tsx:6]:
```suggestion
export const CancelingIcon = forwardRef<ElementRef<typeof SpinnerIcon>, IconSVGProps>(function CancelingIcon(
```

File: libs/shared/ui/src/lib/components/icon/icons-status/unknow.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import { forwardRef } from 'react'
+import { twMerge } from 'tailwind-merge'
+import { type IconSVGProps } from '../icon'
+
+export const UnknowIcon = forwardRef<SVGSVGElement, IconSVGProps>(function UnknowIcon(
```
Comment (ctjhoa) [inline libs/shared/ui/src/lib/components/icon/icons-status/unknow.tsx:5]: `UnknowIcon` -> `UnknownIcon`
also rename the file itself

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/shared/ui/src/lib/components/icon/icons-status/deployed.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import { forwardRef } from 'react'
+import { twMerge } from 'tailwind-merge'
+import { type IconSVGProps } from '../icon'
+
+export const DeployedIcon = forwardRef<SVGSVGElement, IconSVGProps>(({ className = '', ...props }, forwardedRef) => {
```
Comment (ctjhoa) [inline libs/shared/ui/src/lib/components/icon/icons-status/deployed.tsx:5]: This one still has no name

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/shared/ui/src/lib/components/icon/icons-status/unknow.tsx (hunk 7 above)
Comment (RemiBonnet) [reply libs/shared/ui/src/lib/components/icon/icons-status/unknow.tsx:5]: nice catch

File: libs/shared/ui/src/lib/components/icon/icons-status/canceling.tsx (hunk 6 above)
Comment (RemiBonnet) [reply libs/shared/ui/src/lib/components/icon/icons-status/canceling.tsx:6]: Already try but I can't do that 

Capture d’écran 2023-09-19 à 14 26 00

File: libs/shared/ui/src/lib/components/icon/icons-status/build-error.tsx (hunk 5 above)
Comment (RemiBonnet) [reply libs/shared/ui/src/lib/components/icon/icons-status/build-error.tsx:5]: Already try but I can't do that 

Capture d’écran 2023-09-19 à 14 26 00

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**